i18n.py

```
@@ -1,30 +1,88 @@
+        
+        logging.info(f"Loaded translations for language: {language}")
+        return translations
     except FileNotFoundError:
-        print(
-            f"Warning: Translation file for {language} not found. Falling back to English."
-        )
-        # Fallback to English if the requested language file is not found
-        fallback_path = os.path.join(current_dir, "i18n", "en.yml")
-        with open(fallback_path, "r", encoding="utf-8") as file:
-            return yaml.safe_load(file)
+        raise TranslationKeyError(f"Translation file for {language} not found. Looked in: {file_path}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)